architecture/platforms/versal_acap.rst

```diff
@@ -0,0 +1,125 @@
+The OP-TEE support for the AMD/Xilinx Versal Adaptive Compute Acceleration Platform (`Versal ACAP`_) delegates most of its functionality to the `PLM firmware`_ executing in the MicroBlaze processor.
+In the case of cryptographic operations, those operations or ciphers or keys not supported by the hardware will be routed to the `Libtomcrypt`_ software based implementation.
+
+The services offered by PLM firmware are decided at build time: it is therefore important that the PLM firmware build configuration enables the services that OP-TEE will require.
+
+As an example, if OP-TEE is configured to generate a hardware unique key, it  will need access to the PLM Physical Unclonable Function and NVM services.
+
+.. note::
+   Communication between OP-TEE and the PLM uses the IPI mailbox controller being the IPI used selectable via ``CFG_VERSAL_MBOX_IPI_ID``.
+
+As described in freely available documentation for the Versal ACAP `boot-flow`_, the BootROM handles loading the PLM, while the PLM will handle loading the rest of the images including OP-TEE.
+
+A reference BIF file supporting an OP-TEE instance capable of loading the FPGA pdi can be seen below. In this example OP-TEE should be configured with ``CFG_DT=y`` and ``CFG_DT_ADDR=0x00001000``.
+If enabled the platform expects the FPGA bitstream at 0x40000000; the location is configurable using ``CFG_VERSAL_FPGA_DDR_ADDR``.
+
+.. code-block:: none
+
+	ROM_image:
+	{
+		image {
+                      { type=bootimage, file=vpl_gen_fixed.pdi }
+	              { type=bootloader, file=plm.elf }
+	              { core=psm, file=psmfw.elf }
+	        }
+	        image {
+	              id = 0x1c000000, name=apu_subsystem
+	              { type=raw, load=0x00001000, file=versal-vck190-revA-x-ebm-01-revA.dtb }
+	              { type=raw, load=0x40000000, file=fpga.pdi }
+	              { core=a72-0, exception_level=el-3, trustzone, file=bl31.elf }
+ 	              { core=a72-0, exception_level=el-2, file=u-boot.elf }
+	              { core=a72-0, exception_level=el-1, trustzone, load=0x60000000, startup=0x60000000, file=tee-raw.bin }
+	         }
+         }
+
+
+To build the boot-able image AMD/Xilinx uses the `bootgen tool`_; this tool aggregates all the different images in a single binary.
+
+The available configuration options depend on the architecture.
+
+In the case of Versal ACAP using the previously mentioned BIF file, the boot.bin generation would be as follows:
+
+.. code-block:: none
+		
+        $ bootgen -arch versal -image file.bif -o boot.bin
+   
+
+Cryptographic driver
+********************
+
+The Versal ACAP cryptography driver rests on the PLM's `xilsecure`_ service.
+It provides hardware assisted support for:
+
+    1. SHA3-384
+    2. RSA 2048, 4096   
+    3. ECC sign/verify
+    4. AES-GCM   
+
+Other drivers
+*************
+
+The ``Versal ACAP eFUSE`` driver uses the PLM `xilnvm`_ service.
+Access to certain eFuses require specific PLM configuration flags not selectable at run-time.
```

Comment:
Not quite true. All eFUSEs are accessible but how they are used is what is defined by the configuration. Explain how the PUF eFUSEs can be used for storage of the PUF helper data or can be used a user eFUSEs.

Reply:
@nathan-menhorn no, this is 100% true.
And no, the user should know how to use the PUF eFuses.

Reply:
@ldts not sure what you mean as the only configuration macros in the xilnvm library for eFUSEs is using the PUF eFUSEs as user data or using them for PUF helper data.
https://github.com/Xilinx/embeddedsw/blob/master/lib/sw_services/xilnvm/src/server/xnvm_efuse.c#L233-L239
https://github.com/Xilinx/embeddedsw/blob/master/lib/sw_services/xilnvm/src/server/xnvm_efuse.c#L265-L267

Reply:
I mean exactly what I said: that the PLM has the capability of disabling/compile out access to eFuses. And that the PLM does not offer an interface for OP-TEE to query about the eFuse capability. 

From this follows that if the workflow requires access to those eFuses, whoever is putting together the release must make sure the PLM has those eFuses enabled (ie, OP-TEE cant do it)